src/Landmark.java

```diff
@@ -1,13 +1,62 @@
+import org.json.simple.JSONObject;
+
 public class Landmark {
     final String name;
-    final boolean isCity;
+     boolean isCity;
+     boolean isRiver;
+     long id;
+     String s1;
+     String s2;
+     String s3;
+     int nextPoint1;
+     int p1Miles;
+     int nextPoint2;
+     int p2Miles;
+     int riverDepth;
+     int riverWidth;
 
-    public Landmark(String name, boolean isCity, int distance) {
+    public Landmark(String name, boolean isCity, boolean isRiver, int distance) {
         this.name = name;
         this.isCity = isCity;
+        this.isRiver = isRiver;
         this.distance = distance;
     }
 
+    /*
+    *   "willemettevalley":{
+    "name": "The Willemette Valley",
+    "id": 18,
+    "isCity": false,
+    "isRiver": false,
+    "s1": "",
+    "s2":"",
+    "s3":"",
+    "nextPointID": 0,
+    "p1miles": 0,
+    "nextPoint2": "",
+    "p2miles":"",
+    "riverDepth": "",
+    "riverWidth": ""
+    * */
+    public Landmark(JSONObject obj) {
+        this.id = (Long) obj.get("id");
```

Comment:
I had this same issue and I recommend using `Utils.castInt()` to cast from a `long` to an `int`.

Reply:
Utilized the castInt() function. Fixed